### ISSUE TITLE
Can't apply for ball/involvement at same time

### ISSUE THREAD

**Issue by NotRaia** [YaManicKill/flairhq#241]
Haven't gotten to verify this yet, but someone said they applied for involvement flair and then a ball and it seems their first app for the involvement got overwritten. Not sure if it'd be a lot of trouble to make them able to do both at the same time

**Comment by NotRaia** [comment]
![screen shot 2015-06-11 at 3 40 49 pm](https://cloud.githubusercontent.com/assets/8943230/8120088/418045c0-1050-11e5-903c-7e67efbdd77c.png)

Actually, I don't think this is an issue anymore. Maybe something else happened to the other guy.

**Comment by YaManicKill** [comment]
So, this isn't an issue? I'll have a look this evening and see if it *should* be an issue or not. I can't remember how I did this.

**Comment by YaManicKill** [comment]
Yeah, pretty sure this isn't something that happens. Binning. If it comes up again, reopen.

**Comment by not-an-aardvark** [comment]
I finally figured out what was going on here.

When a user is given involvement flair, the 'sparkles' are added on to whatever flair is currently cached by FlairHQ. (For example, if a user has `dreamball` flair and successfully applies for involvement flair, their resulting css class is `dreamball1`.)

Suppose user applies for another flair at the same time (say, `cherishball`), and that application is approved first. Then the user will have `cherishball` flair on reddit, but (due to a bug) their flair on FlairHQ will still be cached as `dreamball` until the user visits the page again

Then if the involvement flair app is approved, the user's new flair is based on FlairHQ's cached `dreamball` css class, so the flair gets set to `dreamball1` instead of `cherishball1`. To a user who can't see the flair apps, this creates the appearance that the Cherish Ball app was never actually approved, and simply disappeared.

(Note that if the involvement flair app is approved first, the user's flair ends up as `cherishball`, creating the involvement that the involvement flair app was never approved.)

Anyway, this was resolved by updating a user's cached flair whenever an app is approved.
